models/problems/int_problem_results.sql

```diff
@@ -2,65 +2,28 @@
 -- contains either the first successful attempt
 -- or the most recent unsuccessful attempt
 
-
--- find the timestamp of the earliest successful response
--- this will be used to pick the xAPI event corresponding to that submission
-with successful_responses as (
-    select
-        org,
-        course_key,
-        problem_id,
-        actor_id,
-        min(emission_time) as first_success_at
-    from
-        {{ ref('fact_problem_responses') }}
-    where
-        -- clickhouse throws an error when shortening this to `where success`
-        success = true
-    group by
-        org,
-        course_key,
-        problem_id,
-        actor_id
-),
--- for all learners who did not submit a successful response,
--- find the timestamp of the most recent unsuccessful response
-unsuccessful_responses as (
-    select
-        org,
-        course_key,
-        problem_id,
-        actor_id,
-        max(emission_time) as last_response_at
-    from
-        {{ ref('fact_problem_responses') }}
-    where
-        actor_id not in (select distinct actor_id from successful_responses)
-    group by
-        org,
-        course_key,
-        problem_id,
-        actor_id
+with outcomes as (
+    select emission_time,
+           org,
+           course_key,
+           problem_id,
+           actor_id,
+           success,
+           first_value(success) over (partition by course_key,
+                                                   problem_id,
+                                                   actor_id
+                                      order by success desc) as was_successful
+    from problem_responses
 ),
--- combine result sets for successful and unsuccessful problem submissions
 responses as (
-    select
-        org,
-        course_key,
-        problem_id,
-        actor_id,
-        first_success_at as emission_time
-    from
-        successful_responses
-    union all
-    select
-        org,
-        course_key,
-        problem_id,
-        actor_id,
-        last_response_at as emission_time
-    from
-        unsuccessful_responses
+    select org, course_key, problem_id, actor_id,
+    if(
+        was_successful,
+        minIf(emission_time, was_successful),
+        maxIf(emission_time, not was_successful)
```

Comment:
Is this double if necessary?

```suggestion
    if(
        was_successful,
        min(emission_time),
        max(emission_time)
```